infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/UniqueWorkspacePVCStrategy.java

```diff
@@ -76,8 +90,13 @@
 public class UniqueWorkspacePVCStrategy implements WorkspaceVolumesStrategy {
 
   private static final Logger LOG = LoggerFactory.getLogger(UniqueWorkspacePVCStrategy.class);
+
   public static final String UNIQUE_STRATEGY = "unique";
 
+  // property of PersistentVolumeClaim#getAdditionalProperties that indicates that PVC is
+  // provisioned by Che Server but is not user-defined
+  private static final String CHE_PROVISIONED_PVC_PROPERTY = "CHE_PROVISIONED";
```

Comment:
The naming of `CHE_PROVISIONED_PVC_PROPERTY` could be more clear -- perhaps something like `EXISTING_CHE_PVC`. It's used to differentiate between PVCs that exist on the cluster and those that are in only the environment, and that's not immediately clear.

Reply:
I think it may be removed after implementing your refactor proposal https://github.com/eclipse/che/pull/12335#discussion_r248075444

Reply:
The confusing part of its use is that starting a workspace a second time will pull the already created user-defined PVC into the env during `fillInExistingPVCs`, which will have the `CHE_PROVISIONED_PVC_PROPERTY` tag. Then, `groupByVolumes` will match this PVC with the volume used for the user defined one, and then `processUserDefinedPVCs` will rewrite and remove the user defined PVC (and instead use the already-provisioned one). 

The confusion for me comes from the fact that the `CHE_PROVISIONED_PVC_PROPERTY` tag will be applied to both e.g. the `che-logs` PVC as well as any user-defined PVCs (the userdata PVC in your sample workspace config).

I'm fine leaving this work out of the PR if there's more refactoring to be done, as it would probably be easier to review at that point anyways, though.